app.py

```diff
@@ -71,38 +70,51 @@ class Cache:
 PREDICTIONS_URL = 'https://api.511.org/transit/StopMonitoring'
 cache = Cache([], '')
 def update_cache():
-    agency_to_stop_suffix = {
-        "BA": "BART",
-    }
     new_stops = []
 
     # send post request for each agency's stop(s)
+    for stop in stops:
+        # check for stop grouping
+        if stop.get("group_name"):
+            group_name = stop.get("group_name")
+            stop_info = get_stop_predictions(stop.get("stops"), group_name)
+            new_stops.append(stop_info)
+            
+        # otherwise, get predictions for individual stop
+        else:
+            stop_name = add_suffix_to_name(stop)
+            stop_info = get_stop_predictions([stop], stop_name)
+            new_stops.append(stop_info)
+
+    cache.stops = new_stops
+    cache.updated_at = datetime.datetime.now(datetime.timezone.utc)
+
+def get_stop_predictions(stops, stop_name):
+    unique_buses: typing.Dict[str, Prediction] = collections.defaultdict(lambda: Prediction("", collections.defaultdict(list)))
+    stop_codes = []
+
     for stop in stops:
         agency = stop.get('operator')
-        stopCode = stop.get('id')
-        stopName= stop.get('name')
-        # We add a suffix to some stop names to add context.
-        # For example, the BART stop called "Milpitas" becomes
-        # "Milpitas BART"
-        if (agency in agency_to_stop_suffix):
-            stopName = stopName + " " + agency_to_stop_suffix.get(agency)
+        stop_code = stop.get('id')
+        stop_codes.append(stop_code)
 
         params = {
             'api_key': API_KEY,
             'agency': agency,
-            'stopCode': stopCode,
+            'stopCode': stop_code,
             'format': 'json'
         }
 
         with MetricsHandler.api_latency.time():
             response = requests.get(PREDICTIONS_URL, params=params)
         logging.debug(f'511`s API response code was {response.status_code}')
+
         # unsuccessful request to 511 API
         if response.status_code != 200:
             MetricsHandler.api_response_codes.labels(response.status_code).inc() 
             logging.error(f"not parsing response because response code was {response.status_code}")
             time.sleep(10)
-            continue
+            return
```

Comment:
why did we change this?

Reply:
`continue` caused an error when I tried doing it a different way but I forgot to change it back

resolved